extensions/amp-video/0.1/video-cache.js

```diff
@@ -33,13 +37,17 @@ import {user} from '../../../src/log';
 export function fetchCachedSources(videoEl, win) {
   if (
     !extensionScriptInNode(win, 'amp-cache-url', '0.1') ||
-    !videoEl.querySelector('source[src]').getAttribute('src')
+    !(
+      videoEl.getAttribute('src') ||
+      videoEl.querySelector('source[src]').getAttribute('src')
```

Comment:
Can you guarantee that `videoEl.querySelector('source[src]')` matches something?